### PR TITLE
#6 General tags for metrics

### LINKED ISSUE #6: Global tags
What is about global/common tags? For example, we have few microservices and one prometheus server. But we want to filter metrics by app name. It will be awesome to add app name tag to any metric. 
How I see this feature. We define global label in configuration
```ruby
Yabeda.configure do
  # declare metrics
  global_label :app_name, value: 'awesome_app'
end 
``` 
It will be add tag `app_name` with value `awesome_app` to each metrics.
What is your opinion about it?

### PR DESCRIPTION
**Pull request by asusikov**
### What do this PR do?

- added ability to set default tags what will be added to each metric

Closes #6

### CODE REVIEW

File: spec/yabeda/dsl/class_methods_spec.rb [hunk 1]
```diff
@@ -106,4 +106,16 @@
       it("defines method on root object") { is_expected.to be_a(Yabeda::Histogram) }
     end
   end
+
+  describe ".general_tag" do
+    subject { Yabeda.general_tags }
+
```
Comment (Envek) [inline spec/yabeda/dsl/class_methods_spec.rb]: Let's add assertion for default behavior here:
```suggestion

it { is_expected.to eq({}) }

```

File: lib/yabeda/version.rb [hunk 2]
```diff
@@ -1,5 +1,5 @@
 # frozen_string_literal: true
 
 module Yabeda
-  VERSION = "0.1.3"
+  VERSION = "0.1.4"
```
Comment (Envek) [inline lib/yabeda/version.rb]: Let's not change versions in PR. I think it worth to change it to 0.2.0 maybe :smiley: , but I will do it when I will release new version to RubyGems

File: spec/yabeda/tags_spec.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+      it { is_expected.to eq({ controller: 'foo' }) }
+    end
+
+    context 'when general tags are set' do
+      before do
+        Yabeda.configure do
+          general_tag :environment, 'test'
+        end
+      end
+
```
Comment (Envek) [inline spec/yabeda/tags_spec.rb]: Let's add spec for merge behavior (I think it is useful):


```suggestion

context "when current tags are the same as general tags" do
  let(:tags) { { environment: 'production' } }

  it { is_expected.to eq({ environment: 'production' }) }
end

```

File: lib/yabeda/version.rb (hunk 2 above)
Comment (asusikov) [reply lib/yabeda/version.rb]: Make sence :) I'll revert it

File: spec/yabeda/dsl/class_methods_spec.rb (hunk 1 above)
Comment (asusikov) [reply spec/yabeda/dsl/class_methods_spec.rb]: I add it to `spec/yabeda_spec.rb`. Should I move it to this file and keep it in `yabeda_spec`?

File: spec/yabeda/tags_spec.rb (hunk 3 above)
Comment (asusikov) [reply spec/yabeda/tags_spec.rb]: Agree - it's great case to add spec

**Comment by asusikov** [comment]
> What do you think about changing name to `default_tags` maybe?
I don't have any cons for it :) Will do
> Also, please squash commits with related changes (e.g. one commit with all about the feature and one with not related specs).
How I remember there is separate button for squash commits when merge PR, not like in GitLab - there is checkbox for it. If yes, I'm going to press it - merge and squash button :)

File: spec/yabeda/dsl/class_methods_spec.rb (hunk 1 above)
Comment (Envek) [reply spec/yabeda/dsl/class_methods_spec.rb]: Ah, now I see. No, then it is not needed anymore

**Comment by Envek** [comment]
Yes, there is separate “Squash and merge” button and I can press it too :-)

File: spec/yabeda/tags_spec.rb (hunk 3 above)
Comment (asusikov) [reply spec/yabeda/tags_spec.rb]: Done

File: lib/yabeda/version.rb (hunk 2 above)
Comment (asusikov) [reply lib/yabeda/version.rb]: Done

**Comment by asusikov** [comment]
@Envek I've done with your requests

**Comment by Envek** [comment]
Thanks! I will merge and release it tomorrow.

**Comment by Envek** [comment]
Released as 0.3.0. Thank you very much!

**Comment by asusikov** [comment]
Awesome!

**Pull request merged**